lib/client.js

```diff
@@ -1397,7 +1397,7 @@ function write (client, request) {
 
     socket.cork()
     socket.write(`${header}content-length: ${contentLength}\r\n\r\n`, 'ascii')
-    socket.write(body)
+    socket.write(body, () => notifyBodyChunkSent(request, contentLength, contentLength))
```

Comment:
Could you cache this closure at a higher level? We should not allocate it every time.

Moreover, we should not allocate it at all if it's not needed by the user.